heka/sandbox/decoders/http_edge_decoder.lua

```diff
@@ -173,6 +174,19 @@ function process_message()
     main_msg.Fields.geoCountry = get_geo_country(xff, remote_addr)
 
     -- Send new message along.
-    inject_message(main_msg)
-    return 0
+    local main_status, main_err = pcall(inject_message, main_msg)
+    local error_message = ""
+    local return_code = 0
+    if not landfill_status then
```

Comment:
The idea was to try both "inject" operations to minimize any data loss, so the error reporting is delayed until the end of processing.